pkg/ingester/ingester_v2.go

```diff
@@ -405,13 +417,23 @@ func (i *Ingester) v2Push(ctx context.Context, req *client.WriteRequest) (*clien
 				}
 			}
 
+			cause := errors.Cause(err)
+			if cause == storage.ErrOutOfBounds && s.TimestampMs > db.Head().MaxTime()-i.cfg.TSDBConfig.BackfillLimit.Milliseconds() {
+				fmt.Println(s.TimestampMs, db.Head().MaxTime(), i.cfg.TSDBConfig.BackfillLimit.Milliseconds())
+				err := i.v2BackfillPush(userID, ts.Labels, s)
```

Comment:
It's not feasible to have 1 TSDB appender commit for every single sample. What I would do instead is:
1. While iterating `for _, s := range ts.Samples {`, I would keep a buffer with all the samples that can be backfilled
2. Below, we partition these backfill samples into buckets and we append all samples at once for each bucket

Reply:
That sounds like a good idea. I will make changes accordingly.